prov/util/src/util_av.c

```diff
@@ -535,11 +534,16 @@ int ofi_av_init_lightweight(struct util_domain *domain, const struct fi_av_attr
 	 */
 	av->context = context;
 	av->domain = domain;
+	av->prov = domain->prov;
 
-
-	ep_list_lock_type = ofi_progress_lock_type(av->domain->threading,
+	lock_type = ofi_progress_lock_type(av->domain->threading,
 						   av->domain->control_progress);
```

Comment:
Please fix the alignment caused by the change.

Reply:
Also I think the lock selection here is actually wrong. The progress lock type is more for the completion devices and the AV doesn't fall into that category. Here's the definition of the progress_lock_type:
```
static inline enum ofi_lock_type
ofi_progress_lock_type(enum fi_threading threading, enum fi_progress control)
{
	return (threading == FI_THREAD_DOMAIN || threading == FI_THREAD_COMPLETION) &&
		control == FI_PROGRESS_CONTROL_UNIFIED ? OFI_LOCK_NOOP : OFI_LOCK_MUTEX;
}
```
The AV still should get locked for thread completion. It can only be optimized out for thread domain.

Reply:
@aingerson thanks, I misunderstood FI_THREAD_COMPLETION. I will fix

Is this the reason for Intel CI failure?

Reply:
hard to say for sure but maybe. the failure is a hang on MPICH verbs-rxm on a multi threaded test. MPICH I think uses FI_THREAD_COMPLETION so definitely a chance there's a race because it optimizes the lock out even though it needs it